packages/venia-ui/lib/components/Dialog/dialog.css

```diff
@@ -0,0 +1,198 @@
+.mask {
+    /* The mask takes up and masks the entire screen. */
+    position: fixed;
+    top: 0;
+    left: 0;
+    height: 100vh;
+    width: 100vw;
+    background-color: rgb(var(--venia-grey-darker));
+    opacity: 0.7;
+
+    /* The Mask sits on top of all other content. */
+    z-index: 3;
+
+    /* The Mask is hidden by default. */
+    visibility: hidden;
+}
+
+.mask_open {
+    composes: mask;
+
+    /* The Mask changes to being visible. */
+    visibility: visible;
+}
+
+.root {
+    /* The Dialog takes up the entire screen and sits on top of the mask. */
+    position: fixed;
+    top: 0;
+    left: 0;
+    height: 100vh;
+    width: 100vw;
+    background-color: transparent;
+    z-index: 4;
+
+    /* Set pointer-events to none to allow the mask to be clicked. */
+    pointer-events: none;
+
+    /*
+     *  The Dialog is a grid container with its container
+     *  horizontally and vertically centered.
+     */
+    display: grid;
+    grid-template-rows: 1fr auto 1fr;
+    grid-template-columns: 1fr auto 1fr;
+    grid-template-areas:
+        '. ......... .'
+        '. container .'
+        '. ......... .';
+
+    /* The Dialog is hidden by default. */
+    opacity: 0;
+    visibility: hidden;
+
+    /* The Dialog animates to being closed. */
+    transition-duration: 192ms;
+    transition-timing-function: var(--venia-anim-out);
+    transition-property: opacity, visibility;
+}
+
+.root_open {
+    composes: root;
+
+    /* The Dialog gets a shadow when open. */
+    box-shadow: 1px 0 rgb(var(--venia-border));
+
+    /* The Dialog animates to being open. */
+    opacity: 1;
+    visibility: visible;
+    transition-duration: 224ms;
+    transition-timing-function: var(--venia-anim-in);
+}
+
+.container {
+    /* The container is the only grid item in the Dialog grid. */
+    grid-area: container;
+    background-color: rgb(var(--venia-background-color));
+
+    /* It sets maximum sizes so its body can handle overflow. */
+    height: 100%;
+    max-height: 90vh;
+    /* Minimum keeps a 16:9 aspect ratio. */
+    min-height: 416px;
+    width: 740px;
+
+    /* Nothing is allowed to overflow container itself. */
+    overflow: hidden;
+
+    /* Container is itself a grid container for its children. */
+    display: grid;
+    grid-template-rows: auto 1fr;
+    grid-template-areas:
+        'header'
+        'body';
+
+    /* Container can be the target of pointer events. */
+    pointer-events: auto;
+}
+
+/*
+ *  Dialog Header styles.
+ */
+
+.header {
+    grid-area: header;
+
+    border-bottom: 1px solid rgb(var(--venia-border));
+
+    /* The Header is itself a grid container for its children. */
+    display: grid;
+    grid-auto-flow: column;
+    padding: 0.875rem;
+}
+
+.headerText {
+    /* Vertically align the title in the middle. */
+    align-self: center;
```

Comment:
Since this isn't really an exception—all items in the header are more or less vertically centered—you should probably just set this on `.header` instead, with `align-items` or `align-content`.

Reply:
Fixed in https://github.com/magento/pwa-studio/pull/2365/commits/d5841abb26d7a014b3bf9d50bdd5fe38c4dc0b77.